internal/core/workers/operation_execution_worker/operation_execution_worker_test.go

```diff
@@ -490,4 +522,95 @@ func TestSchedulerOperationsExecutionLoop(t *testing.T) {
 		err := workerService.Start(context.Background())
 		require.NoError(t, err)
 	})
+
+	t.Run("when healthcontroller ticks, call operation manager creating a new health_controller operation", func(t *testing.T) {
+		duration, err := time.ParseDuration("1ms")
+		require.NoError(t, err)
+
+		mockCtrl := gomock.NewController(t)
+
+		config := mockconfig.NewMockConfig(mockCtrl)
+		config.EXPECT().GetDuration(gomock.Any()).Return(duration)
+
+		operationName := "test_operation"
+		operationDefinition := mockoperation.NewMockDefinition(mockCtrl)
+		operationExecutor := mockoperation.NewMockExecutor(mockCtrl)
+		operationManager := mock.NewMockOperationManager(mockCtrl)
+
+		defFunc := func() operations.Definition { return operationDefinition }
+		definitionConstructors := operations.NewDefinitionConstructors()
+		definitionConstructors[operationName] = defFunc
+
+		scheduler := &entities.Scheduler{Name: "random-scheduler"}
+
+		executors := map[string]operations.Executor{}
+		executors[operationName] = operationExecutor
+		workerService := NewOperationExecutionWorker(scheduler, workers.ProvideWorkerOptions(config, operationManager, executors, nil, nil))
+		pendingOpsChan := make(chan string)
+
+		operationManager.EXPECT().PendingOperationsChan(gomock.Any(), gomock.Any()).Return(pendingOpsChan)
+		operationManager.EXPECT().CreateOperation(gomock.Any(), gomock.Any(), gomock.Any()).Do(func(_ context.Context, schedulerName string, op operations.Definition) {
+			require.Equal(t, scheduler.Name, schedulerName)
+			require.IsType(t, &healthcontroller.SchedulerHealthControllerDefinition{}, op)
+		}).Return(&operation.Operation{}, nil).AnyTimes()
```

Comment:
No need to use Do() here, right?

```suggestion
operationManager.EXPECT().CreateOperation(gomock.Any(), scheduler.Name, &healthcontroller.SchedulerHealthControllerDefinition{}).Return(&operation.Operation{}, nil).AnyTimes()

```

Reply:
Besides that, what about using **MaxTimes(5)** instead of **AnyTimes()**? we wait 5ms for the test to run and the period is 1ms, so it should never trigger the healthcontroller more than 5 times.